src/cwe_checker_lib/src/analysis/function_signature/mod.rs

```diff
@@ -266,6 +278,111 @@ impl FunctionSignature {
+                    false => std::cmp::Ordering::Less,
+                }
+            })
+            .tuple_windows()
+        {
+            if let Ok(((address, size), log)) =
+                get_bounds_intersecting_stack_arg(first_arg, second_arg)
+            {
+                let merged_arg = Arg::Stack {
+                    address,
+                    size: ByteSize::from(size),
+                    data_type: first_arg.get_data_type(),
+                };
+                merged_args.insert(merged_arg, first_pattern.merge(second_pattern));
+                removed_args.insert(first_arg.clone());
+                removed_args.insert(second_arg.clone());
+                logs.extend(log.clone());
+            }
+        }
+        for arg in removed_args {
+            self.parameters.remove(&arg);
+        }
+        self.parameters.extend(merged_args);
+        Ok(logs)
+    }
+}
+
+/// Merges two stack parameters and returns address and size.
+/// Also returns a message, if one argument is not a subset of the other one.
+///
+/// Returns `Err` if `first_arg` or `second_arg`:
+/// * are not `Arg::Stack` types
+/// * do not have the same `Datatype`
+/// * return `Err` on `Arg::eval_stack_offset()`
+/// * do not intersect
+fn get_bounds_intersecting_stack_arg(
+    first_arg: &Arg,
+    second_arg: &Arg,
+) -> Result<((Expression, u64), Vec<String>), Error> {
+    if let (
+        Arg::Stack {
+            data_type: first_datatype,
+            size: first_size,
+            address: first_address,
+        },
+        Arg::Stack {
+            data_type: second_datatype,
+            size: second_size,
+            ..
+        },
+    ) = (first_arg, second_arg)
+    {
+        let first_arg_offset = first_arg.eval_stack_offset()?.try_to_u64()?;
+        let first_arg_size = u64::from(*first_size);
+        let second_arg_offset = second_arg.eval_stack_offset()?.try_to_u64()?;
+        let second_arg_size = u64::from(*second_size);
+
+        if first_datatype == second_datatype {
+            let mut logs = vec![];
+            let first_arg_upper_bound = first_arg_offset + first_arg_size;
+            // Check if they intersect
+            if first_arg_upper_bound >= second_arg_offset {
+                let second_arg_upper_bound = second_arg_offset + second_arg_size;
+                // Check if subset
+                if second_arg_upper_bound <= first_arg_upper_bound {
+                } else {
+                    // second arg is not a subset, but intersects
+                    logs.push(
+                        "Merged a stack parameter, that intersect another but is no subset"
+                            .to_string(),
+                    );
+                }
+                let merged_size = second_arg_upper_bound - first_arg_offset;
+                return Ok(((first_address.clone(), merged_size), logs));
+            } else {
+                return Err(anyhow!("Args do not intersect"));
```

Comment:
You do not actually use the error cases, do you? If not, you could change the return type of the function to an `Option` instead of a `Result`.